tests/test_analysis_utils.py

```diff
@@ -89,6 +97,22 @@ def setUp(self):
             im = layered_image(str(i), self.dim_x, self.dim_y, self.noise_level, self.variance, time, self.p)
             self.imlist.append(im)
         self.stack = image_stack(self.imlist)
+        
+        # Set up old_results object for analysis_utils.PostProcess
+        self.num_curves = 4
+        # First 3 passing indices
+        psi_curves = [np.array([1.0 + (x / 100) for x in range(20)]) for _ in range(self.num_curves - 1)]
+        phi_curves = [np.array([1.0 + (y / 100) for y in range(20)]) for _ in range(self.num_curves - 1)]
+        # Failing index
+        psi_curves.append(np.array([-200.0 - (100.0 * z) for z in range(20)]))
```

Comment:
We can get a better test if we define the last vector as a series of regular values with two outliers and save a vector of indices that pass the filter:

```suggestion
        # Failing index (generate a list of psi values such that the elements 2 and 14 are filtered
        # by sigmaG filtering.
        failing_psi = [0.0 + (z / 100) for z in range(20)]
        failing_psi[14] = -100.0
        failing_psi[2] = 100.0
        psi_curves.append(np.array(failing_psi))
        phi_curves.append(np.array([1.0 for _ in range(20)]))

        psi_good_indices = [z for z in range(20)]
        psi_good_indices.remove(14)
        psi_good_indices.remove(2)
        self.good_indices = np.array(psi_good_indices)
```

The tests below would then change from:

`self.assertEqual(res[self.num_curves - 1][1], [-1])`

to

```        
self.assertEqual(len(res[self.num_curves - 1][1]), len(self.good_indices))
for index in range(len(res[self.num_curves - 1][1])):
    self.assertEqual(res[self.num_curves - 1][1][index], self.good_indices[index])
```